### ISSUE TITLE
[rllib] memory specs in ray.init not respected

### ISSUE THREAD

**Issue by justinkterry** [ray-project/ray#6785]
`ray.init(redis_max_memory=int(1e10), object_store_memory=int(30000000000))`

turns into 

`2020-01-13 18:03:51,899	INFO resource_spec.py:216 -- Starting Ray with 15.48 GiB memory available for workers and up to 27.94 GiB for objects. You can adjust these settings with ray.init(memory=<bytes>, object_store_memory=<bytes>).`

**Comment by justinkterry** [comment]
I forgot to include a key part of the error:
In addition to it starting with different specs than I specified, it creates a ton of of actors and I shortly get this error:

`ray.memory_monitor.RayOutOfMemoryError: More than 95% of the memory on node Prophet is used (62.73 / 62.85 GB). The top 10 memory consumers are:`

In other words, ray is seemingly ignoring it's own made up limits?

**Comment by letmaik** [comment]
I have the same issue using tune:
```py
    ray.init(
        memory=5 * 1024 * 1024 * 1024, # 5 GB
        object_store_memory=200 * 1024 * 1024, # 200 MB
        driver_object_store_memory=100 * 1024 * 1024, # 100 MB
        redis_max_memory=100 * 1024 * 1024, # 100 MB
        local_mode=True,
        logging_level=logging.INFO,
    )
```

becomes 

> 2020-03-05 09:07:15,643 INFO resource_spec.py:212 -- Starting Ray with 43.12 GiB memory available for workers and up to 18.63 GiB for objects. You can adjust these settings with ray.init(memory=<bytes>, object_store_memory=<bytes>).

**Comment by letmaik** [comment]
Note that when not using local_mode with the same parameters I get:
> 2020-03-05 09:30:28,152 INFO resource_spec.py:212 -- Starting Ray with 68.21 GiB memory available for workers and up to 0.2 GiB for objects. You can adjust these settings with ray.init(memory=<bytes>, object_store_memory=<bytes>).

So it seems `memory` is never taken into account, but the object store memory is in this case.
